packages/flutter/lib/src/widgets/editable_text.dart

```diff
@@ -1533,9 +1536,15 @@ class EditableTextState extends State<EditableText> with AutomaticKeepAliveClien
   TextInputConnection? _textInputConnection;
   TextSelectionOverlay? _selectionOverlay;
 
+  // The source of the most recent selection change.
+  //
+  // Changing the selection programmatically does not update
+  // _selectionChangedCause.
+  SelectionChangedCause? _selectionChangedCause;
+
   ScrollController? _scrollController;
 
-  late AnimationController _cursorBlinkOpacityController;
+  late final AnimationController _cursorBlinkOpacityController = AnimationController(vsync: this, duration: _fadeDuration);
```

Comment:
I think we missed this but it should be disposed in the dispose method.
same for _floatingCursorResetController. I think the existing format is cleaner, it is weird the we give a default value in here and then attach listener in initState

Reply:
These should be fixed in https://github.com/flutter/flutter/pull/87973

Reply:
moved the `addListener` to the lazy initializer. My initial goal was to mark the property as final. I'll do the same for other animation controllers in #87973.